### ISSUE TITLE
primary keys from DatabaseMetaData.getPrimaryKeys  are case-insensitive 

### ISSUE THREAD

**Issue by downgoon** [xerial/sqlite-jdbc#219]
## a table in ``default.db``

``` sql
CREATE TABLE `FRIEND` (
    `USER_id`   INTEGER NOT NULL,
    `FRIE_id`   INTEGER NOT NULL,
    `degree`    INTEGER,
    primary key ( `USER_id`, `FRIE_id` )
);
```

## testing code

``` java
import java.sql.Connection;
import java.sql.DatabaseMetaData;
import java.sql.DriverManager;
import java.sql.ResultSet;

public class SqliteJdbcAutoIncr {

	public static void main(String[] args) throws Exception {
		Class.forName("org.sqlite.JDBC");
		// Connection conn = DriverManager.getConnection("jdbc:sqlite:auto_incr.db");
		Connection conn = DriverManager.getConnection("jdbc:sqlite:default.db");

		DatabaseMetaData dbmeta = conn.getMetaData();

		// Listing Tables
		String catalog = null;
		String schemaPattern = null;
		String tableNamePattern = null;
		String[] types = null;

		ResultSet tables = dbmeta.getTables(catalog, schemaPattern, tableNamePattern, types);
		while (tables.next()) { // for each table

			final String tableName = tables.getString(3);

			// Listing Columns in a Table
			String columnNamePattern = null;
			ResultSet columns = dbmeta.getColumns(catalog, schemaPattern, tableName, columnNamePattern);

			System.out.println("table: " + tableName);
			System.out.println("----------");
			
			while (columns.next()) {
				String columnName = columns.getString(4);
				// int columnType = columns.getInt(5);

				boolean isAutoIncrement = columns.getBoolean("IS_AUTOINCREMENT");
				String autoIncString = columns.getString("IS_AUTOINCREMENT");

				System.out.println(
						String.format("\tcol: %s isAutoIncrement: %s %s", columnName, isAutoIncrement, autoIncString));
			}

			System.out.println();
			
			String schema = null;
			ResultSet keys = dbmeta.getPrimaryKeys(catalog, schema, tableName);
			while (keys.next()) {
				String keyName = keys.getString(4); // COLUMN_NAME
				int keySeq = keys.getInt("KEY_SEQ");
				
				System.out.println("primary key: " + keyName + ", " + keySeq);
			}


		}

	}

}

```

## output result

```
table: FRIEND
----------
	col: USER_id isAutoIncrement: false NO
	col: FRIE_id isAutoIncrement: false NO
	col: degree isAutoIncrement: false NO

primary key: `frie_id`, 1
primary key: `user_id`, 0
table: sqlite_sequence
----------
	col: name isAutoIncrement: false NO
	col: seq isAutoIncrement: false NO
```

## case-insensitive ``getPrimaryKeys``

field in sql script | dbmeta.getColumns | dbmeta.getPrimaryKeys
------------ | ------------- |  -------------
`USER_id` | USER_id | `user_id` 
`FRIE_id` | FRIE_id | `frie_id` 
`degree` | degree | - 

```
tow issues

- case-insensitive:  USER_id is expected, rather than its lower case
- char ` not trimmed: `user_id`  -> user_id
```

**Comment by sualeh** [comment]
@xerial - please close this issue, since it is fixed with your merge.

**Comment by sualeh** [comment]
@xerial - please close this issue, since it is fixed with your merge of #246

**Comment by sualeh** [comment]
Fixed in 3.20.1
